apps/events/api/register_attendance_serializer.py

```diff
@@ -0,0 +1,159 @@
+import logging
+
+from django.db import IntegrityError
+from django.db.models import Q
+from rest_framework import serializers, status
+from rest_framework.serializers import ValidationError
+
+from apps.authentication.models import OnlineUser as User
+from apps.events.constants import AttendStatus
+from apps.events.models import Attendee, Event
+
+logger = logging.getLogger(__name__)
+
```

Comment:
This is just so damned complex that i moved it to its own file.
We have functionality to replace it, but regme uses this _hard_ and would have to be completly re-written to make that work

Reply:
I suggest we rather make a `serializers`-module, split every serializer into it's own file (or whatever makes most sense) and re-export everything from  `__init__.py`. I think that would be a much cleaner solution.
I also think we should do what I suggested to the models.
I know you have said earlier that that isn't "the django way", but it would be a lot more work should we split this up into different apps, which I don't think is something we want, and I think it would be highly beneficial to split these gigantic files (1000+ lines, like come on) compared to the effort it takes.

Reply:
I can agree on splitting some of it up, like the serializers. Not that it can be a bit more modular, like splitting them up by use, such as a file for `RuleBundles` and all the `Rules`.
I don't actually think we should re-export the serializers from `__init__.py`, its OK to just import them from where they are.

I've had another look into splitting up models, and the Django Docs actually write about it: https://docs.djangoproject.com/en/3.0/topics/db/models/#organizing-models-in-a-package
It should be very doable, but I still maintain that they should still be somewhat grouped, not all in their own file.
I also think we have to do it all the time, we have plenty of apps that have few enough models for it not to be a problem at all

Reply:
@FluidSense Correct for this file, but some things have been added, and some things have been changed. I just tried to comment in the places which were mostly old code